hooks/tk-multi-publish2/basic/publish_session_geometry.py

```diff
@@ -0,0 +1,574 @@
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        """
+
+        publisher = self.parent
+
+        # get the data for the publish that was just created in SG
+        publish_data = item.properties["sg_publish_data"]
+
+        # ensure conflicting publishes have their status cleared
+        publisher.util.clear_status_for_conflicting_publishes(
+            item.context, publish_data)
+
+        self.logger.info(
+            "Cleared the status of all previous, conflicting publishes")
+
+        publish_path = item.properties["publish_info"]["publish_path"]
+
+        path = item.properties["publish_file_path"]
+
+        self.logger.info(
+            "Publish created for file: %s" % (publish_path,),
+            extra={
+                "action_show_in_shotgun": {
+                    "label": "Show Publish",
+                    "tooltip": "Open the Publish in Shotgun.",
+                    "entity": publish_data
+                }
+            }
+        )
+
+
+    def _get_next_version_info(self, path, settings):
+        """
+        Return the next version of the supplied path.
+
+        If templates are configured, use template logic. Otherwise, fall back to
+        the zero configuration, path_info hook logic.
+
+        :param str path: A path with a version number.
+        :param settings: Configured settings for this plugin.
+
+        :return: A tuple of the form::
+
+            # the first item is the supplied path with the version bumped by 1
+            # the second item is the new version number
+            (next_version_path, version)
+        """
+
+        publisher = self.parent
+
+        # get the configured work file template
+        work_template = self._get_template("Work file Template", settings)
+
+        # set these so we can check to see if the template logic applied
+        next_version_path = None
+        version = None
+
+        if work_template:
+            # if the work file template matches, we'll get parsed fields
+            fields = work_template.validate_and_get_fields(path)
+
+            if fields and "version" in fields:
+                # template matched. bump the version number and re-apply to
+                # the template
+                fields["version"] += 1
+                next_version_path = work_template.apply_fields(fields)
+                version = fields["version"]
+
+        if not next_version_path and not version:
+            # fall back to the "zero config" logic
+            next_version_path = publisher.util.get_next_version_path(path)
+            version = publisher.util.get_version_number(next_version_path)
+
+        return (next_version_path, version)
+
+    def _get_publish_info(self, path, settings):
```

Comment:
Yeah, it stinks that this method may be repeated in our plugins. Perhaps we should think about moving this into `publisher.util`. Let me ponder.

Reply:
No worries/rush